internal/libs/autofile/autofile.go

```diff
@@ -114,6 +126,13 @@ func (af *AutoFile) closeFile() (err error) {
 	af.mtx.Lock()
 	defer af.mtx.Unlock()
 
+	defer func() {
+		// Otherwise reset .hasWrites if there was no error.
+		if err == nil {
```

Comment:
I think we need to `err = file.Close()` so that an error there will impact this value?